### PR TITLE
[DEVX-3608] Add custom description tag

### PR DESCRIPTION
**Pull request by fabianrbz**
## Description

The `<meta description>` and `<og:description>` tags will now use the same value.
If a `frontmatter` is present, the content will be the value of `@frontmatter['meta_description]` or `@frontmatter['description']` providing the correct value for all the docs and custom pages.

For the landing page, `@frontmatter` is not defined, and in that case, the value comes from `config/header_meta.yml`

### CODE REVIEW

File: lib/nexmo_developer/app/presenters/head.rb [hunk 1]
```diff
@@ -1,66 +1,96 @@
...
-  def initialize(items: nil)
-    @items = items
+    def description
+      @description ||= from_frontmatter || @config.fetch('description') do
+        raise "You must provide a 'description' parameter in header_meta.yml"
+      end
+    end
+
+    def from_frontmatter
+      @frontmatter && (@frontmatter['meta_description'] || @frontmatter['description'])
```
Comment (bencgreenberg) [inline lib/nexmo_developer/app/presenters/head.rb:15]: Should we use kebab-case here for `meta_description` to match the rest of the parameters in the header meta YAML file?

File: lib/nexmo_developer/app/presenters/head.rb [hunk 2]
```diff
@@ -1,66 +1,96 @@
...
+  def title
+    @title ||= config.fetch('title') do
+      raise "You must provide a 'title' parameter in header_meta.yml"
+    end
   end
 
-  def open_config(config)
-    File.open(config)
+  def description
+    @description ||= Description.new(config: config, frontmatter: frontmatter).description
```
Comment (bencgreenberg) [inline lib/nexmo_developer/app/presenters/head.rb:35]: This means the `description` parameter must be present, right? I tried with only `meta_description` in the frontmatter and the app raised an exception:

```
ActionView::Template::Error (You must provide a 'description' parameter in header_meta.yml):
     9:   <% end %>
    10:   <title><%= page_title %></title>
    11: 
    12:   <% if head.description %>
    13:     <meta name="description" content="<%= head.description %>">
    14:   <% end %>
    15: 
```

File: lib/nexmo_developer/app/presenters/head.rb [hunk 3]
```diff
@@ -1,66 +1,96 @@
...
+    config_path = "#{Rails.configuration.docs_base_path}/config/header_meta.yml"
+    raise 'You must provide a config/header_meta.yml file in your documentation path.' unless File.exist?(config_path)
+
+    @config = YAML.safe_load(File.open(config_path))
+  end
 
-    "meta/#{file_name}"
+  def validate_files_presence!
+    %i[favicon favicon_32_squared manifest safari_pinned_tab mstile_144_squared apple_touch_icon og_image].each do |file|
+      raise "You must provide an #{send(file).sub('meta/', '')} file inside the public/meta directory" unless File.exist?("#{Rails.configuration.docs_base_path}/public/#{send(file)}")
```
Comment (bencgreenberg) [inline lib/nexmo_developer/app/presenters/head.rb:93]:
```suggestion
      raise "You must provide a #{send(file).sub('meta/', '')} file inside the public/meta directory" unless File.exist?("#{Rails.configuration.docs_base_path}/public/#{send(file)}")
```

File: lib/nexmo_developer/app/presenters/head.rb (hunk 1 above)
Comment (fabianrbz) [reply lib/nexmo_developer/app/presenters/head.rb:15]: we can, but it wouldn't be backwards compatible :/, so I left it as it was

Comment (bencgreenberg) [reply lib/nexmo_developer/app/presenters/head.rb:15]: Good point, yes let's leave it then

File: lib/nexmo_developer/app/presenters/head.rb (hunk 3 above)
Comment (fabianrbz) [reply lib/nexmo_developer/app/presenters/head.rb:93]: I'm quite certain that we are not scrapping the logs for this exception, so it is ok to update its text 👍

File: lib/nexmo_developer/app/presenters/head.rb (hunk 2 above)
Comment (fabianrbz) [reply lib/nexmo_developer/app/presenters/head.rb:35]: yes, it's the same behavior as before
https://github.com/Nexmo/station/blob/master/lib/nexmo_developer/app/presenters/head.rb#L55-L59

**Pull request merged**